### DISCUSSION TITLE
How to remove stuck instance snapshot

### DISCUSSION THREAD

**Discussion by SviridoffA** [apache/cloudstack#9874] [Q&A]
Hi everyone! Can someone help me? I use cloudstack 4.19.1.2, and linstor as a primary storage. I know that vmsnapshot not suported for linstor, but when I'm testing it was unknown to me. Now I have vmsnapshot which  stuck in reverting state. And I nothing can do with this. I tryed to delete it by api, cmk or ui, but its always return "Can't delete the vm snapshotshot 2 due to it is not in Created or Error, or Expunging State". Can someone tell me how I can do it corectly?

**Comment by SviridoffA** [comment]
Since I didn't get any clue, I decided to use the hard method. I found a table in the database named vm_snapshots and made two updates to it. I changed state from 'Reverting' to 'Expunging', and changed removed from Null to date. Now this snapshot not showing in UI and I can make operations with VM, its not blocked anymore. Not sure its a right way. But i dont have another ideas.

**Reply by DaanHoogland** [reply]
@SviridoffA , I am sure there should be a better way, but this is a solid workaround. If we understood what happened, we could create a more automated fix.

**Reply by SviridoffA** [reply]
@DaanHoogland Unfortunaly, I can't remember how to reproduce this. It's happend more than two month ago, and all this time it was in reverting state. I remembered about this snapshot? only when decided to delete related vm. I sure it was only disk snapshot without memory, and i tryed restor vm from it or tryed to delete this snapshot or chain of snapshots but it stuck. And I didnt found any related files in primary or secondary storage.
